instrumentation/aws_sdk/lib/opentelemetry/instrumentation/aws_sdk/instrumentation.rb

```diff
@@ -46,14 +47,42 @@ def require_dependencies
           require_relative 'messaging_helper'
         end
 
-        def add_plugin(*targets)
+        def add_plugins(*targets)
           targets.each { |klass| klass.add_plugin(AwsSdk::Plugin) }
         end
 
         def loaded_constants
+          if gem_version >= CURRENT_MAJOR_VERSION
+            load_v3_constants
+          else
+            load_legacy_constants
```

Comment:
Is there a reason the dynamic approach doesn't work in V2?  I think it should and may be better to avoid checking against (and defining the current major version).

Reply:
Thanks to CI (🥹✨) failing on V2 cases. I think my assumption that V2 Clients inherits from `Seahorse::Client::Base` was wrong. What it actually happens is that V2 Clients inherits from `Aws::Client` and THEN, inherits from `Seahorse::Client::Base`. So I have made a minor adjustment to cover this.